service/src/main/java/uk/nhs/adaptors/gp2gp/common/storageconnectors/StorageConnector.java

```diff
@@ -0,0 +1,12 @@
+package uk.nhs.adaptors.gp2gp.common.storageconnectors;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.List;
+
+public interface StorageConnector {
+    void uploadToStorage(InputStream is, String filename) throws IOException;
```

Comment:
We might consider defining a StorageConnectorException to wrap any backend-specific errors that are thrown